app/scaffold/DOM.js

```diff
@@ -167,6 +167,7 @@ export function configurePageLogger(logger) {
                 typeof JSON.stringify === "function"
             ) {
                 output += JSON.stringify(arg, replacer, 4);
+                hideElement(alertBoxElement);
```

Comment:
In case you are confused as to why this is here, every time the iron core api returns, it returns and object that is put into console.log. So whenever an object is passed into console.log, then that means that the request was received, and we can hide the loading message.

Reply:
Can't you move this so there's a call in each `.then()` handler of the associated function below? For example, couldn't you call this on line 190 in the scaffold.js file? You'd have to make a call for each function, but at least that way you'd have more control over when you actually want to hide the loader.

Reply:
The problem with this is that adding and removing away team members doesn't have an obvious promise in which we can hide the message.

Reply:
Those operations do have Promises, we just aren't doing anything special when they're done executing. For example, the code to add people to the group is currently

```
if (this.addAwayTeamMembersFn(crewmemberIdsToAdd)) {
  DOM.alertBox("Adding Member...");
  model.addAwayTeamMembers(crewmemberIdsToAdd);
  DOM.moveItemsBetweenSelects(DOM.fromElement, DOM.toElement);
}
```

We could pretty easily change that to

```
const addMembersPromise = this.addAwayTeamMembersFn(crewmemberIdsToAdd);
if(!addMembersPromise){
  return;
}
DOM.alertBox("Adding Member...");
model.addAwayTeamMembers(crewmemberIdsToAdd);
DOM.moveItemsBetweenSelects(DOM.fromElement, DOM.toElement);
addMembersPromise.then(() => {
  hideElement(alertBoxElement);
  console.log("Successfully added new members to away team");
})
```

Or something similar. The code would end up being pretty similar for the remove members function too.